### DISCUSSION TITLE
Can't get basic layout to work

### DISCUSSION THREAD

**Discussion by A-Walrus** [motion-canvas/motion-canvas#1067] [Q&A] [answered]
Trying to have a row of squares with text centered within them. Can't seem to get it to work. I tried
```ts
  view.add(
    <Layout layout  gap={20} padding={20}>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt>0</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt>1</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt>2</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt>3</Txt></Rect>
    </Layout>
  );
```
which results in:
![image](https://github.com/motion-canvas/motion-canvas/assets/58790821/95ebc249-0a81-4936-91bf-93d12391af44)

tried messing around with `layout={false}` on the children, and some other properties but nothing seems to work.

Any help appreciated, thanks!

**Comment by RithwikAnand** [comment]
You might want to try this:
```    
    <Layout layout gap={20} padding={20}>
      <Rect width={100} height={100} fill="gray" radius={15} layout justifyContent="center" alignItems="center"><Txt>0</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15} layout justifyContent="center" alignItems="center"><Txt>1</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15} layout justifyContent="center" alignItems="center"><Txt>2</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15} layout justifyContent="center" alignItems="center"><Txt>3</Txt></Rect>
    </Layout>
```

**Reply by A-Walrus** [reply]
Thanks that works!
I'm still wondering if there is a nicer way to this, since if the `Rect`s aren't in a layout the `Txt` is just naturally centered.
Shouldn't there be a way to tell the layout to stop propagating and treat the `Rect`s like normal?

**Reply by aarthificial** [reply]
@A-Walrus 
You can do that by setting `layout={false}` on the `Txt` itself. This will position it in the center of the `Rect` as if there was no layout.

**Reply by A-Walrus** [reply]
That doesn't seem to work:
```ts
export default makeScene2D(function*(view) {
  view.add(
    <Layout layout gap={20} padding={20}>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt layout={false}>0</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt layout={false}>1</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt layout={false}>2</Txt></Rect>
      <Rect width={100} height={100} fill="gray" radius={15}><Txt layout={false}>3</Txt></Rect>
    </Layout>
  );
});
```
![image](https://github.com/user-attachments/assets/bea5f9c2-1cba-4e42-8715-92d2e133cb16)

could this be a bug?

**Reply by aarthificial** [reply, marked as answer]
Apologies, I forgot that `Txt` nodes force-enable layout because they use it to lay out the text inside them.
Setting `layout={false}` would work if you had a different node inside, say a `Circle`. 

Well then, `justifyContent` would be the way to go in this case.
Alternatively, you could wrap the `Txt` in it's own `Layout` and disable it this way:
```tsx
<Rect width={100} height={100} fill="gray" radius={15}>
  <Layout layout={false}>
    <Txt>0</Txt>
  </Layout>
</Rect>
```
But that's probably more verbose than `justifyContent` + `alignItems`
